src/reports/client/main.cpp

```diff
@@ -0,0 +1,200 @@
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <iterator>
+#include <thread>
+#include <vector>
+
+#ifdef HAVE_MPI
+#include <mpi.h>
+#endif
+
+#include <bbp/sonata/reports.h>
+#include <reports/utils/logger.hpp>
+
+struct Neuron {
+    int node_id;
+    std::string kind;  // soma / element
+    std::vector<double> voltages;
+};
+
+const static double dt = 0.1;
+const static double tstart = 0.0;
+const static double tstop = 0.3;
+
+void generate_spikes(const std::vector<uint64_t>& nodeids,
+                     std::vector<double>& spike_timestamps,
+                     std::vector<int>& spike_node_ids,
+                     int seed,
+                     int max_size) {
+    // Generate 10,30,50,70,90 spikes
+    int num_spikes = (10 + (20 * seed)) % 100;
+    spike_timestamps.reserve(num_spikes);
+    spike_node_ids.reserve(num_spikes);
+    for (int i = 0; i < num_spikes; i++) {
+        // timestamp between tstart and tstop
+        double timestamp = tstart + static_cast<double>(0.5 + seed) /
+                                        (static_cast<double>(max_size / (tstop - tstart)));
+        // get an index to the nodeids
+        int index = seed % nodeids.size();
+        int gid = nodeids[index];
+        spike_timestamps.push_back(timestamp);
+        spike_node_ids.push_back(gid);
+    }
+}
+
+void generate_elements(Neuron& neuron, int seed) {
+    // 50+-5 elements
+    int num_elements = 50 + ((seed % 10) - 5);
+    if (neuron.kind == "soma") {
+        num_elements = 1;
+    }
+    neuron.voltages.reserve(num_elements);
+    for (int j = 0; j < num_elements; j++) {
+        neuron.voltages.push_back(seed % 10);
+    }
+}
+
+std::vector<uint64_t> generate_data(std::vector<Neuron>& neurons,
+                                    const std::string& kind,
+                                    int seed) {
+    std::vector<uint64_t> nodeids;
+    // Each gid starts with the rank*10 (i.e. rank 5 will have gids: 51, 52, 53...)
+    uint64_t nextgid = 1 + seed * 10;
+    uint32_t c_id = 0;
```

Comment:
I was only checking for the sonata_report target. After adding it there were a couple of variables not used.
Since I will move it to tests, does it make sense to check? the unit tests are not checked i believe...

Reply:
I'm generally more permissive with style in unittests; I'm just surprised that this wasn't triggering an `unused variable` because I would expect it to have `-Wall -Werror` applied.  That's why I think it's worth investigating why it didn't break compilation.